### PR TITLE
Restructure e2e tests for future cases

### PR DESCRIPTION
**Pull request by aliok**
Fixes #

## Proposed Changes

- Restructure e2e tests for future cases
- Created separate directories for components
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Comment by aliok** [comment]
/retest

Strimzi operator fails. ...interesting

File: .github/workflows/kind-e2e.yaml [hunk 1]
```diff
@@ -58,12 +61,35 @@ jobs:
             channel-auth-scenario: "SASL_SSL"
           - test-suite: ./test/e2e
             channel-auth-scenario: "SASL_PLAIN"
+
+          - test-suite: ./test/e2e_broker
+            channel-auth-scenario: "SSL"
+          - test-suite: ./test/e2e_broker
+            channel-auth-scenario: "SASL_SSL"
+          - test-suite: ./test/e2e_broker
+            channel-auth-scenario: "SASL_PLAIN"
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply .github/workflows/kind-e2e.yaml:70]: Same question for source and sink suites.

[the remark this answers is not in the document]
Comment (aliok) [reply .github/workflows/kind-e2e.yaml:70]: Those are exclude cases

[the remark this answers is not in the document]
Comment (pierDipi) [reply .github/workflows/kind-e2e.yaml:70]: what do you mean?

[the remark this answers is not in the document]
Comment (pierDipi) [reply .github/workflows/kind-e2e.yaml:70]: Ah, I see

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**